File: Arrays_and_Hashing/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    
    let hashMap = {};
    const strlen = nums.length;
    
    for (let i = 0; i < strlen; i++) {
        
        let hashTarget = target - nums[i];
        
        if (hashMap[hashTarget] === undefined) {
            hashMap[nums[i]] = i;
        }
        else {
            return [hashMap[hashTarget], i];
        }
        
    }

};

File: Arrays_and_Hashing/TwoSum.go
func twoSum(nums []int, target int) []int {
    hashMap := make(map[int]int)

    for i, val := range nums {
        newTarg := target - val

        if value, isFound := hashMap[newTarg]; isFound {
            return []int{value, i}
        }

        hashMap[val] = i
    }
    return []int{}
}
